src/actions/Entities/VaccinePrescriptionActions.js

```diff
@@ -163,13 +163,41 @@ const createPrescription = (
   });
 };
 
-const createRefusalNameNote = name => {
-  const [patientEvent] = UIDatabase.objects('PatientEvent').filtered('code == "RV"');
+const createVaccinationNameNote = (
+  patient,
+  prescription,
+  refused,
+  bonusDose,
+  vaccinator,
+  selectedBatch
+) => {
+  const [patientEvent] = UIDatabase.objects('PatientEvent').filtered('code == "vaccination"');
 
   if (!patientEvent) return;
 
   const id = generateUUID();
-  const newNameNote = { id, name, patientEvent, entryDate: new Date() };
+  const data = {
+    refused,
+    bonusDose,
+    vaccine: selectedBatch?.itemName,
+    vaccineCode: selectedBatch?.itemCode,
+    batch: selectedBatch?.batch,
+    expiry: selectedBatch?.expiryDate,
+    vaccinator: vaccinator?.displayString,
+    extra: {
+      prescription: prescription?.toJSON(),
+      vaccinator: vaccinator.toJSON(),
+      patient: patient.toJSON(),
+    },
```

Comment:
🤷

Reply:
😄  I tested first - and saw all this on the server! 
sure, why not..